android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java

```diff
@@ -668,8 +670,9 @@ public static boolean registerLexicalModel(HashMap<String, String> lexicalModelI
       return false;
     }
 
-    // Use single quotes in JSONobject for javascript call
+    // Escape single quotes, and then convert double quotes to single quotes for javascript call
     String model = String.valueOf(modelObj);
+    model = model.replaceAll("\'", "\\\\'"); // Double-escaped-backslash b/c regex.
```

Comment:
Is that supposed to replace `\'` with `\\'`? If so, you need to have:
```suggestion
    model = model.replaceAll("\\'", "\\\\'"); // Double-escaped-backslash b/c regex.
```

Reply:
The code is lifted from KMKeyboard.java ll.350-351 for escaping single-quoted keyboard and language names
```java
    // Escape single-quoted names for javascript call
    keyboardName = keyboardName.replaceAll("\'", "\\\\'"); // Double-escaped-backslash b/c regex.
    languageName = languageName.replaceAll("\'", "\\\\'");
```
It looks to replace a literal `'` with a double-escaped backslash